lib/mongo/socket.rb

```diff
@@ -169,7 +169,22 @@ def eof?
     private
 
     def read_from_socket(length)
-      @socket.read(length) || String.new
+      # Block on data to read for timeout seconds
+      # using the suggested implementation of http://www.ruby-doc.org/core-2.1.3/Kernel.html#method-i-select
+      # to work with SSL connections and some OSs like Ubuntu, CentOS and Red Hat
+      time_left = timeout.to_f
+      data = ""
```

Comment:
I think you should use `String.new` here to emphasize the fact that it has to be mutable and be ready for the Ruby 3.0 immutable string literals.